src/components/SpeechRec.vue

```diff
@@ -0,0 +1,183 @@
+
+      
+    </button>
+    <button v-if="ttsActive" 
+            @click="stopTTS"     
+            type="button" 
+            class="tts-button custom-icon">
+      <span v-if="ttsStopSymbol" v-html="ttsStopSymbol"></span>
+      <ttsIcon v-else class="custom-icon" aria-hidden="true"/>
+    </button>
+  </div>
+</template>
+
+<script>
+import { mapState, mapGetters } from 'vuex';
+import AsrIcon from '../icons/asr.vue';
+import AsrMuteIcon from '../icons/asr-mute.vue';
+import ttsIcon from '../icons/tts.vue';
+
+export default {
+  components: {
+    AsrIcon,
+    AsrMuteIcon,
+    ttsIcon
+  },
+  props: {
+    userInputFieldId: String,
+  },
+  data() {
+    return {
+      transcribing: false,
+      buttonPressed: false,
+      recognition: null,
+      alertMessage: this.$t('message.webspeech_not_supported'),
+      lastResult: ''
+    };
+  },
+
+  computed: {
+    ...mapState(['asrRecordSymbol', 'ttsStopSymbol', 'asrPauseSymbol']),
+    ...mapGetters({
+      asrActive: 'asrActive',
+      ttsActive: 'ttsActive'
+    }),
+    
+  },
+
+  methods: {
+    startTranscription() {
+      this.buttonPressed = true;
+      
+      // Before creating a new SpeechRecognition cleanup
+      if (this.recognition) {
+        this.recognition.stop(); 
+        this.recognition = null; 
+      }      
+      const SpeechRecognition = window.SpeechRecognition || window.webkitSpeechRecognition;
+
+      if (typeof SpeechRecognition === "undefined") {
+        alert(this.alertMessage);
+        return;
+      }
+      if (!this.transcribing) {
+        this.recognition = new SpeechRecognition();
+        this.recognition.continuous = true;
+        const isMobile = /Android|webOS|iPhone|iPad|iPod|BlackBerry|IEMobile|Opera Mini/i.test(navigator.userAgent);
+        this.recognition.interimResults = !isMobile;
+        this.recognition.addEventListener('result', this.resultHandler);
+        this.recognition.addEventListener('end', this.endHandler);
+        this.recognition.start();
+        this.transcribing = true;
+        this.$emit('transcribing', this.transcribing);
+      }
+    },
+
+    stopASR() {
+      this.buttonPressed = false;
+      if (this.recognition) {
+        this.recognition.stop();
+      }
+      this.transcribing = false;
+      this.$emit('transcribing', this.transcribing);      
+    },
+
+    readTranscription(text) {
+         const utterance = new SpeechSynthesisUtterance(text);
+         window.speechSynthesis.speak(utterance);
+         this.$forceUpdate();
+    },
+    
+    handleMouseLeave() {
+      if (this.transcribing) {
+        this.stopASR();
+      }
+    },
+
+    stopTTS() {
+      window.speechSynthesis.cancel();
+    },
+
+    resultHandler(event) {
+      this.$emit('transcription', event.results[event.results.length - 1][0].transcript);      
+      if (event.results[event.results.length - 1].isFinal) {
+        this.$emit('transcriptionComplete', event.results);
```

Comment:
UserInput.vue is still expecting (but not using!) a String value here - but this is sending the results object
We should align the types